platform/ios/src/MGLMapView.mm

```diff
@@ -6924,4 +6929,11 @@ - (void)setShowsHeading:(BOOL)showsHeading
     self.showsUserHeadingIndicator = showsHeading;
 }
 
+#pragma mark - MGLNetworkEventDelegate
```

Comment:
`MGLMapView` shouldn't conform to `MGLNetworkEventDelegate`. (See https://github.com/mapbox/mapbox-gl-native/issues/13789#issuecomment-457226452)

We want to put the decision to track downloads in application code, so we should find a mechanism for the developer to provide a `MGLNetworkEventDelegate`.

And since `-[MGLMapboxEvents pushEvent:withAttributes:]` isn't public, we should discuss how we expose that ability too. Perhaps it's as simple as adding a method to `MGLMapView`. This seems to be what happens on Android ([`onPerformanceEvent`](https://github.com/mapbox/mapbox-gl-native/pull/13795/files)) - though I don't speak Android. @tobrun can you confirm how it works?

I'm also wondering if there should be a public protocol(s) for public event methods, event names and attributes that we expose? /cc @rclee for input.

Reply:
We allow the user to get an instance from Telemetry through our Mapbox.java configuration object on this we expose pushing a performance event

Reply:
I think a protocol makes sense for exposing that method. We don't have to do this now but perhaps it would be good to provide the user some sort of opt-in and then do all of this event generation and collection without further developer input. 

cc @alfwatt @andrlee for visibility

Reply:
> `MGLMapView` shouldn't conform to `MGLNetworkEventDelegate`.

According to #13794 where we want to trace tile download performance I made the following considerations:
- The code that deals with network request is the class `http_file_source.mm` (and is the best place to actually measure the time without being affected by other process) which is part of the `mbgl-filesource` target. The `MGLNetworkConfiguration` seemed like the best place to deal with the management for the events (registering start dates, stops, cancels) and call the `pushEvent` method. But the restrictions for the target which has no access to the telemetry library (and I'm not sure it's possible to rule out configuration for macos) made me think that I could send a notification to send this event or I could make the `mapView` send it's own performance events.  

What you are saying is that the developers will control creating and sending the tile download performance event as well?

I agree on adding a generic performance event class, but I think that should be done in the telemetry library because otherwise I will duplicate the same functionality as https://github.com/mapbox/mapbox-navigation-ios/pull/1954

cc @rclee

Reply:
> What you are saying is that the developers will control creating and sending the tile download performance event as well?

I like how you have it set up - that if a delegate method is implemented then the SDK creates the event. We then alert the developer, and they decide what to do with the event (we give them some method to push it). It might just be a case of moving where that delegate implementation lives; from `MGLMapView` to application code.

Reply:
> I agree on adding a generic performance event class, but I think that should be done in the telemetry library [...]

Agreed. Does this block the PR? Or can we just move the event class into the telemetry lib later? /cc @rclee

Reply:
@julianrex i don't think we should block this PR. We have intentionally removed whitelisting mechanism to enable faster iteration cycles of other sdk features. We'd like to avoid releasing new versions of telem library every time this event has to change.

Reply:
@fabian-guerra i realize there's a duplication of effort here between maps and nav sdk but i don't agree that this necessarily belongs to telemetry sdk. imho in ideal world we need performance sdk that hides complexity of perf metrics instrumentation and takes care of perf event creation/management. so i'm okay with adding a predefined event type tag to telem constants but not more than that.